R/backport_linter.R

```diff
@@ -0,0 +1,86 @@
+#' @describeIn linters that checks for usage of unavailable functions. Not reliable for testing r-devel dependencies.
+#' @export
+backport_linter <- function(r_version = getRversion()) {
+  function(source_file) {
+    if (inherits(r_version, "numeric_version")) r_version <- format(r_version)
+    if (r_version < "3.0.0") {
```

Comment:
We should use `numeric_version` comparison here, else it will bite us when R 10.0.0 rolls out ;)

Reply:
haha that can be a problem for our future selves -- after 2050 at the current rate :smile_cat:

I prefer this way because the test below makes the same assumption by doing `r_version < names(backports)`

So really this assumption is backed into `names(backports)` as well. Maybe by R 10.0 R will allow objects to have objects as names like python